crates/vm/levm/src/precompiles.rs

```diff
@@ -846,6 +846,7 @@ const R4: u32 = 63;
 /// v[0..15].  The full modified vector is returned.
 /// Based on https://datatracker.ietf.org/doc/html/rfc7693#section-3.1
 #[allow(clippy::indexing_slicing)]
+#[inline(always)]
 fn g(v: [u64; 16], a: usize, b: usize, c: usize, d: usize, x: u64, y: u64) -> [u64; 16] {
```

Comment:
Can you check what happens if instead of inlining the three functions you change the signature of this one to:
```suggestion
fn g(v: &[u64; 16], a: usize, b: usize, c: usize, d: usize, x: u64, y: u64, res: &mut [u64; 16]) {
```